### PR TITLE
Kzl 642 search result

### PR DESCRIPTION
**Pull request by benoitvidis**
Depends on https://github.com/kuzzleio/sdk-go/pull/218 and future upcoming related prs.*

## What does this PR do?

Following https://github.com/kuzzleio/sdk-go/pull/218; this PR exposes the `SearchResultxx.next` methods to the Cgo layer.

https://github.com/kuzzleio/sdk-go/pull/218 :arrow_left: :large_blue_circle: :arrow_right: https://github.com/kuzzleio/sdk-cpp/pull/18

### CODE REVIEW

File: cgo/kuzzle/c_to_go.go [hunk 1]
```diff
@@ -141,27 +141,106 @@ func cToGoUserRigh(r *C.user_right) *types.UserRights {
...
+		Subcommand:   C.GoString(r.subcommand),
+		Pattern:      C.GoString(r.pattern),
+		Idx:          int(r.idx),
+		Min:          C.GoString(r.min),
+		Max:          C.GoString(r.max),
+		Limit:        C.GoString(r.limit),
+		Count:        int(r.count),
+		Match:        C.GoString(r.match),
+		Reset:        bool(r.reset),
+		IncludeTrash: bool(r.include_trash),
```
Comment (Aschen) [inline cgo/kuzzle/c_to_go.go:183]: Didn't we say we were depreciating this option?

File: cgo/kuzzle/c_to_go.go [hunk 2]
```diff
@@ -141,27 +141,106 @@ func cToGoUserRigh(r *C.user_right) *types.UserRights {
...
+	}
 
-	c, _ := json.Marshal(s.collection)
+	response.Error = types.NewError(C.GoString(r.error), int(r.status))
+	response.Error.Stack = C.GoString(r.stack)
 
-	var documents json.RawMessage
-	d, _ := json.Marshal(s.documents)
-	documents = d
+	response.Result, _ = json.Marshal(r.result)
```
Comment (Aschen) [inline cgo/kuzzle/c_to_go.go:219]: Why no error checking here ?

File: include/kuzzlesdk.h [hunk 3]
```diff
@@ -132,6 +133,8 @@ typedef struct {
     const char *limit;
     unsigned long count;
     const char *match;
+    const bool reset;
+    const bool include_trash;
```
Comment (Aschen) [inline include/kuzzlesdk.h]: Same remark about deprecated option

File: include/kuzzlesdk.h [hunk 4]
```diff
@@ -132,6 +133,8 @@ typedef struct {
     const char *limit;
     unsigned long count;
     const char *match;
+    const bool reset;
```
Comment (Aschen) [inline include/kuzzlesdk.h]: If you are using a `const bool` here (which I approve), why didn't use `const double` `const size_t` `const long` also ?

**Review by Aschen — CHANGES_REQUESTED** [review]

File: cgo/kuzzle/c_to_go.go (hunk 2 above)
Comment (jenow) [reply cgo/kuzzle/c_to_go.go:219]: Kuzzle returns only json response and KuzzleResponse is a response from Kuzzle, it is impossible de get something else than json as a kuzzle response

File: cgo/kuzzle/c_to_go.go (hunk 1 above)
Comment (benoitvidis) [reply cgo/kuzzle/c_to_go.go:183]: I was not aware it was, sorry.
By deprecating, do you mean it should be removed already from all Sdks or only not documented or marked as such?

Comment (benoitvidis) [reply cgo/kuzzle/c_to_go.go:183]: ok, I just checked and this option is used by `count`, `get` and `mget` only.
I agree it has no added-value. I created a ticket to remove it in another set of prs: https://jira.kaliop.net/browse/KZL-803

Comment (Aschen) [reply cgo/kuzzle/c_to_go.go:183]: We should mark it deprecated in the documentation v1 but maybe we could get ride of it directly in documenation v2 and especially in the SDKs

**Review by Aschen — APPROVED** [review]

**Review by jenow — APPROVED** [review]

**Pull request merged**